### PR TITLE
Context API

### PR DESCRIPTION
**Pull request by sebastiandedeyne**

### CODE REVIEW

File: src/BladeXCompiler.php [hunk 1]
```diff
@@ -72,29 +72,37 @@ protected function parseClosingTags(string $viewContents, BladeXComponent $blade
...
 
-        return $this->componentStartString($bladeXComponent, $attributes).$this->componentEndString();
+        return $this->componentStartString($bladeXComponent, $attributes).$this->componentEndString($bladeXComponent);
     }
 
     protected function componentStartString(BladeXComponent $bladeXComponent, string $attributes = ''): string
     {
         $componentAttributeString = "[{$attributes}]";
 
+        if ($bladeXComponent->bladeViewName === 'bladex::context') {
```
Comment (sebastiandedeyne) [inline src/BladeXCompiler.php]: This hack kind of sucks. Maybe `BladeXComponent` is the wrong abstraction now?

Comment (sebastiandedeyne) [reply src/BladeXCompiler.php]: Maybe we can find a way that also gets rid of the `context.blade.php` view.

File: tests/BladeXTest.php [hunk 2]
```diff
@@ -17,9 +17,9 @@ public function it_can_register_a_single_component_with_providing_a_view_and_com
 
         $registeredComponents = BladeX::getRegisteredComponents();
 
-        $this->assertCount(1, $registeredComponents);
-        $this->assertEquals('myView1', $registeredComponents[0]->name);
-        $this->assertEquals('registerDirectoryTest/myView1', $registeredComponents[0]->bladeViewName);
+        $this->assertCount(2, $registeredComponents);
```
Comment (sebastiandedeyne) [inline tests/BladeXTest.php]: Needed to increment all these assertions by one. This was quite tedious, what if we want to add another first-party component in the future?

Comment (freekmurze) [reply tests/BladeXTest.php]: Maybe we can introduce some the concept of component types and not let `BladeX::getRegisteredComponents` return internal ones?

**Comment by freekmurze** [comment]
Magic! Thanks!

**Pull request merged**